src/cargo/core/compiler/mod.rs

```diff
@@ -409,7 +409,17 @@ fn rustc(
                         )
                     },
                 )
-                .map_err(verbose_if_simple_exit_code)
+                .map_err(|e| {
+                    if output_options.errors_seen == 0 {
+                        // If we didn't expect an error, do not require --verbose to fail.
+                        // This is intended to debug
```

Comment:
`verbose_if_simple_exit_code` doesn't do anything with `--verbose` so I am confused by the comment.

(I am also confused by `verbose_if_simple_exit_code`... the comment says that on signals and "abnormal termination", "don't hide the error details" -- but the code and function name seem to make it *more* verbose when it is a "simple" error, i.e. *not* a signal / abnormal termination.

Reply:
Oh, I see... `VerboseError` is an error that is *less* verbose, except if `--verbose` is passed then it is always verbose. That's quite confusing.

Reply:
Agree. Should have to rename it to something. Do you have any suggestion?

Reply:
Maybe something like `SilenceErrorUnlessVerbose`?